### PR TITLE
added test case for expression having @ and +

### PR DESCRIPTION
**Pull request by hiteshbedre**
Covered test case for: https://github.com/joernio/joern/pull/3135
Expression such as: x+@y

### CODE REVIEW

**Comment by hiteshbedre** [comment]
@xavierpinho please have review.

**Review by xavierpinho — APPROVED** [review]

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 1]
```diff
@@ -1187,6 +1187,26 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
...
+        |  end
+        |
+        |  def calculate_x(b)
+        |    x = b+@a
+        |    return x
+        |  end
+        |end
+        |""".stripMargin)
+
+    cpg.identifier("a").dedup.size shouldBe 1
```
[the remark this answers is not in the document]
Comment (DavidBakerEffendi) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1204]: I think that `@a` should be a field access structure, and somewhere on here there could be a modifier to denote if it's a dynamic or static access, which should correspond to `@` and `@@` right?

[the remark this answers is not in the document]
Comment (xavierpinho) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1204]: > I think that `@a` should be a field access structure, and somewhere on here there could be a modifier to denote if it's a dynamic or static access, which should correspond to `@` and `@@` right?

Didn't think of that and sounds perfectly reasonable. Just one question though: being a field access means we still need to have the field `a`, right? For instance, in

```ruby
class MyClass
 @@a = 1
 attr_accessor :a

 def init(a)
   @@a = a
   @a = a
 end
end
```

we would need to have a field for `@@a` and another for `@a`, or am I missing something?

**Comment by hiteshbedre** [comment]
Added one more case for https://github.com/joernio/joern/issues/3199
Case: Code snippet containing unless keyword and regex pattern

**Comment by xavierpinho** [comment]
> Added one more case for #3199 Case: Code snippet containing unless keyword and regex pattern

It would be preferable to have this in another PR, since it's unrelated.

**Comment by hiteshbedre** [comment]
> > Added one more case for #3199 Case: Code snippet containing unless keyword and regex pattern
> 
> It would be preferable to have this in another PR, since it's unrelated.

Added in current PR as pipeline `pr / test-scripts (pull_request)` is getting failed many times.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (DavidBakerEffendi) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1204]: @xavierpinho if the above is valid ruby, then you make a good point here that we should maybe want to keep the `@`

[the remark this answers is not in the document]
Comment (xavierpinho) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1204]: > @xavierpinho Just want to check, is this to be resolved in this PR or another?

Another PR, let's merge this. Just wanted to bring this discussion to the table.

**Pull request merged**